### PR TITLE
Feat/reinitialized network switcher

### PR DESCRIPTION
**Pull request by alvaro-ricotta**
This commit includes an updated related to : 

The direct network switch from MetaMask has been enabled, the logic has been modified to automatically update balances in MM after changing the address, the logic has been modified to force a restart of the connection once the network is changed

https://github.com/orgs/bobanetwork/projects/10/views/1?pane=issue&itemId=37030457
https://github.com/orgs/bobanetwork/projects/10/views/1?pane=issue&itemId=37848069
https://github.com/orgs/bobanetwork/projects/10/views/1?pane=issue&itemId=37848327

### CODE REVIEW

File: packages/boba/gateway/src/containers/modals/dao/old/NewProposalModalOldDao.js [hunk 1]
```diff
@@ -10,154 +10,86 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
-    <Modal open={open} onClose={handleClose} maxWidth="sm" title="Create Proposal">
+    <Modal
+      open={open}
+      onClose={handleClose}
+      maxWidth="sm"
+      title="Create Proposal"
+    >
       <Box>
 
         <Box sx={{ display: 'flex', flexDirection: 'column', gap: '10px' }}>
```
Comment (sk-enya) [inline packages/boba/gateway/src/containers/modals/dao/old/NewProposalModalOldDao.js:95]: As you are working on this component. can you please get rid of importing `Box` from material ui and make use of `styled component` with typescript.

File: packages/boba/gateway/src/hooks/useGasWatcher.ts [hunk 2]
```diff
@@ -50,6 +50,7 @@ const useGasWatcher = () => {
       fetchGas()
 
       if (activeNetwork === NETWORK.ETHEREUM) {
+        console.log('im ethereum')
```
Comment (sk-enya) [inline packages/boba/gateway/src/hooks/useGasWatcher.ts]: remove console please if it's not required

File: packages/boba/gateway/src/hooks/useSwitchChain.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ const useSwitchChain = () => {
   const layer = useSelector(selectLayer())
 
   const switchChain = () => {
+    console.log('switching change')
```
Comment (sk-enya) [inline packages/boba/gateway/src/hooks/useSwitchChain.ts]: one more console please remove

File: packages/boba/gateway/src/hooks/useWalletConnect.ts [hunk 4]
```diff
@@ -48,12 +48,12 @@ export const useWalletConnect = () => {
       const initialized = await networkService.initializeAccount({
         chainIdChanged,
       })
-
       if (initialized === 'nometamask') {
         dispatch(openModal('noMetaMaskModal'))
         return false
       } else if (initialized === 'wrongnetwork') {
-        dispatch(openModal('wrongNetworkModal'))
+        console.log('wrong network account', networkService.account)
```
Comment (sk-enya) [inline packages/boba/gateway/src/hooks/useWalletConnect.ts]: please get rid of console here too.

**Review by sk-enya — CHANGES_REQUESTED** [review]
Please remove console which has been used for debugging purpose.

File: packages/boba/gateway/src/hooks/useWalletConnect.ts (hunk 4 above)
Comment (alvaro-ricotta) [reply packages/boba/gateway/src/hooks/useWalletConnect.ts]: Console log has been removed.

File: packages/boba/gateway/src/hooks/useSwitchChain.ts (hunk 3 above)
Comment (alvaro-ricotta) [reply packages/boba/gateway/src/hooks/useSwitchChain.ts]: Console log has been removed.

File: packages/boba/gateway/src/hooks/useGasWatcher.ts (hunk 2 above)
Comment (alvaro-ricotta) [reply packages/boba/gateway/src/hooks/useGasWatcher.ts]: Console log has been removed.

**Review by sk-enya — APPROVED** [review]
Change looks good, 

Please remove the consoles before merging to develop!

File: packages/boba/gateway/src/services/networkService.js [hunk 5]
```diff
@@ -3676,7 +3686,7 @@ class NetworkService {
   }
 
   async getFS_Info() {
-
+    console.log(this.account)
```
Comment (sk-enya) [inline packages/boba/gateway/src/services/networkService.js]: please remove!

File: packages/boba/gateway/src/hooks/useWalletConnect.ts [hunk 6]
```diff
@@ -48,12 +48,10 @@ export const useWalletConnect = () => {
       const initialized = await networkService.initializeAccount({
         chainIdChanged,
       })
-
       if (initialized === 'nometamask') {
         dispatch(openModal('noMetaMaskModal'))
         return false
       } else if (initialized === 'wrongnetwork') {
-        dispatch(openModal('wrongNetworkModal'))
```
Comment (sk-enya) [inline packages/boba/gateway/src/hooks/useWalletConnect.ts:56]: @alvaro-ricotta why did we remove here, I didn't capture this earlier but it's the issue here.

**Pull request merged**